tests/test_helpers/test_pipe/test_pipe.py

```diff
@@ -137,6 +138,42 @@ async def test_connection_refused(self):
         assert connected is False
 
 
+def make_future(result) -> asyncio.Future:
+    """Make future for value."""
+    f = asyncio.Future()  # type: ignore
+    f.set_result(result)
+    return f
+
+
+@pytest.mark.asyncio
+class TestAEAHelperTCPSocketChannelTLS:
+    """Test that TCPSocketChannelTLS work properly"""
+
+    SERVER_PUB_KEY = (
+        "03e09d7febfc3e3ef4c38321fb10c4303956e65e1d03ead77470941e82428f80d2"
+    )
+    SIGNATURE = b"0D\x02 \x7f\xcf\xa7\xe9~\x1d{\xf1\xb6\"\x98\x1d'\xd2}\x13\xb5\x13\x0c,]}7\xf3\xa6G\x958q_\x89\x08\x02 >\x91\x9e!\xb5\x1f/\xe4\x0c\xab\x9ej\xb7~Z8\x0b\x06Et\x9f\x1b.SZ*Q\xb7\x13\x85\xaa-"
+    SESSION_KEY = b"\x04D\x08\xe1\xca\xfc;\x01(\x0f\xb9&\x92\x12\xe6\x0b\x02\xdc\x082\xa3\xff\x05\x1f#\xeaK\xd3!\xf5\xcc\xcf\x86\x98\x17g\x80_\xc7o;\x9e\x86,O\xbd\xa1bO\x06\x92\x85\x94\x9f\xbf}\xc2\xdd[\xe6AI\x9a\x9c\x92"
```

Comment:
How do we generate all this when the connection is used in an agent?

Reply:
server pub key is from connection configuration
signature generated by libp2p node with private key
session pub key is tmp key generated by libp2p node

Reply:
So the signature (I assume using connection private key) needs to be requested in the context of a cert request? What's the signature of?

Reply:
signature is a session public key value signed with peer private key that can be verified by the peer public key(address)